src/openforms/authentication/contrib/digid_eherkenning_oidc/views.py

```diff
@@ -25,103 +13,23 @@
-            DigiDOIDCAuthentication,
-            EHerkenningBewindvoeringOIDCAuthentication,
-            eHerkenningOIDCAuthentication,
-            get_config_to_plugin,
-        )
-
-        config_to_plugin = get_config_to_plugin()
-        config_class = lookup_config(self.request)
-        assert issubclass(config_class, BaseConfig)
-        plugin = config_to_plugin[config_class]
-
-        match error, error_description, plugin:
-            case (
-                "access_denied",
-                "The user cancelled",
-                DigiDOIDCAuthentication() | DigiDMachtigenOIDCAuthentication(),
-            ):
-                return (DIGID_MESSAGE_PARAMETER, LOGIN_CANCELLED)
-
-            case (
-                "access_denied",
-                "The user cancelled",
-                eHerkenningOIDCAuthentication()
-                | EHerkenningBewindvoeringOIDCAuthentication(),
-            ):
-                eh_message_parameter = EH_MESSAGE_PARAMETER % {
-                    "plugin_id": plugin.identifier.split("_")[0]
-                }
-                return (eh_message_parameter, LOGIN_CANCELLED)
-
-            case _:
-                return (BACKEND_OUTAGE_RESPONSE_PARAMETER, plugin.identifier)
-
-    @property
-    def failure_url(self) -> str:
-        """
-        On failure, redirect to the form with an appropriate error message.
-        """
-        # this is expected to be the auth plugin return url, set by the OIDCInit view
-        plugin_return_url = furl(
-            self._redirect_next or self.get_settings("LOGIN_REDIRECT_URL", "/")
-        )
-        # this URL is expected to have a ?next query param pointing back to the frontend
-        # where the form is rendered/embedded
-        _next = plugin_return_url.args["next"]
-        assert isinstance(_next, str)
-        form_url = furl(_next)
-
-        parameter, problem_code = self.get_error_message_parameters(
-            error=self.request.GET.get("error", ""),
-            error_description=self.request.GET.get("error_description", ""),
-        )
-        form_url.args[parameter] = problem_code
-        return form_url.url
+class OptionsT(TypedDict):
+    pass
 
 
-digid_init = OIDCInit.as_view(
+digid_init = OIDCAuthenticationInitView[OptionsT].as_view(
     config_class=OFDigiDConfig,
     allow_next_from_query=False,
 )
-digid_machtigen_init = OIDCInit.as_view(
+digid_machtigen_init = OIDCAuthenticationInitView[OptionsT].as_view(
     config_class=OFDigiDMachtigenConfig,
     allow_next_from_query=False,
 )
-eherkenning_init = OIDCInit.as_view(
+eherkenning_init = OIDCAuthenticationInitView[OptionsT].as_view(
     config_class=OFEHerkenningConfig,
     allow_next_from_query=False,
 )
-eherkenning_bewindvoering_init = OIDCInit.as_view(
+eherkenning_bewindvoering_init = OIDCAuthenticationInitView[OptionsT].as_view(
     config_class=OFEHerkenningBewindvoeringConfig,
     allow_next_from_query=False,
 )
```

Comment:
passsing a generic typevar in a concrete instantiation doesn't make a lot of sense - you would typically pass the actual plugin-specific option shapes here. But either way there's little point in passing the type var because we don't do anything with the init views ourselves anymore that may consume the `options`, it's all down to django at this point to call the view.